kong/db/dao/plugins/go.lua

```diff
@@ -36,6 +36,47 @@ do
   end
 end
 
+do
+  local pluginserver_proc
+
+  function go.manage_pluginserver()
+    if pluginserver_proc then
+      -- already being managed
+      return
+    end
+
+    if ngx.worker.id() ~= 0 then
+      -- only one manager
+      pluginserver_proc = true
+      return
+    end
+
+    ngx_timer_at(0, function ()
+      local ngx_pipe = require "ngx.pipe"
+
+      while true do
+        kong.log.notice("Starting go-pluginserver")
+        pluginserver_proc = assert(ngx_pipe.spawn({
+          kong.configuration.prefix .. "/bin/go-pluginserver",
+          "-socket", go.socket_path(),
+          "-plugins-directory", kong.configuration.go_plugins_dir,
+        }, {
+          environ = { "PATH=" .. kong.configuration.prefix },
+        }))
+
+        while true do
```

Comment:
We should check to see if `ngx.worker.exiting()` returns `true`, if so then the process should be killed.

Reply:
Initially I intended to add a `plugin.Exit()` method, but crash-only processes are simpler to maintain.  the process is currently killed by the OS when the worker exits, but i guess it's nicer to do it here first.